subsys/debug/gdbstub.c

```diff
@@ -36,7 +36,7 @@ LOG_MODULE_REGISTER(gdbstub);
 #define GDB_ERROR_MEMORY    "E14"
 #define GDB_ERROR_OVERFLOW  "E22"
 
-static bool not_first_start;
+static int not_first_start;
```

Comment:
The problem described doesn't seem right. Maybe it's an toolchain issue?

Reply:
The file is now restored back to the original. I tested it again with the bool data type and it seems to work fine.